src/features/embalm/stepContent/context/CreateSarcophagusContext.tsx

```diff
@@ -0,0 +1,84 @@
+import React, { createContext, ReactNode, useEffect, useState } from 'react';
+import { createEncryptionKeypairAsync } from '../hooks/useCreateEncryptionKeypair';
+import { useSelector } from '../../../../store';
+import { ArchaeologistEncryptedShard } from '../../../../types';
+
+interface CreateSarcophagusContextProps {
```

Comment:
This seems a little strange to me. Like we have created a context to set the `outerKeys`? Maybe I'm missing something, or this can't be contained in a component or a hook?

Reply:
Both the outer key generation and the public keys `useEffect` probably belong in their own hooks -- will get this refactor handled in a followup branch.